File: send_email.py
import os
import requests

def send_email():
    # Captura o e-mail do autor do último commit
    email_recipient = os.popen("git log -1 --pretty=format:'%ae'").read().strip()

    if not email_recipient:
        print("Erro: Não foi possível capturar o e-mail do autor do commit.")
        exit(1)

    # Mailgun API Key e URL
    mailgun_api_key = os.getenv("MAILGUN_API_KEY")  # API Key do Mailgun
    mailgun_domain = os.getenv("MAILGUN_DOMAIN")  # Domínio Mailgun
    mailgun_url = f"https://api.mailgun.net/v3/{mailgun_domain}/messages"

    # Detalhes do e-mail
    from_email = "no-reply04@sandboxb1d4be8e79454fdcb42de699e1f4374a.mailgun.org"  # O e-mail de envio (pode ser qualquer e-mail associado ao Mailgun)
    subject = "CI/CD Pipeline"
    body = "Pipeline executado com sucesso!"

    # Enviar o e-mail usando Mailgun
    response = requests.post(
        mailgun_url,
        auth=("api", mailgun_api_key),
        data={
            "from": from_email,
            "to": email_recipient,
            "subject": subject,
            "text": body
        })

    if response.status_code == 200:
        print(f"E-mail enviado para {email_recipient}")
    else:
        print(f"Erro ao enviar e-mail: {response.status_code} - {response.text}")


if __name__ == '__main__':
    send_email()

File: send_email.js
const nodemailer = require("nodemailer");
require("dotenv").config(); // Para carregar as variáveis de ambiente, se necessário

// Função para criar o transporte de e-mails
function createTransporter() {
    const email = process.env.EMAIL_USERNAME;
    const password = process.env.EMAIL_PASSWORD;

    if (!email || !password) {
        console.error("Erro: Variáveis de ambiente EMAIL_USERNAME ou EMAIL_PASSWORD não estão configuradas.");
        process.exit(1);
    }

    return nodemailer.createTransport({
        service: "gmail", // ou outro serviço SMTP
        auth: {
            user: email,
            pass: password, // Use uma senha de app para segurança
        },
    });
}

// Função para enviar o e-mail
async function sendEmail(recipient) {
    console.log("Iniciando envio de e-mail...");

    const transporter = createTransporter();
    const subject = "CI/CD Pipeline Notification";
    const text = `Olá,

O pipeline foi executado com sucesso!

Status: Concluído com sucesso
Data: ${new Date().toLocaleString()}

Verifique os detalhes no Jenkins.

Atenciosamente,
Equipe CI/CD`;

    try {
        const info = await transporter.sendMail({
            from: `"Jenkins CI/CD" <${process.env.EMAIL_USERNAME}>`,
            to: recipient,
            subject: subject,
            text: text,
        });

        console.log("E-mail enviado com sucesso para %s (ID: %s)", recipient, info.messageId);
    } catch (error) {
        console.error("Erro ao enviar o e-mail:", error);
        process.exit(1);
    }
}

// Captura o destinatário do último commit e envia o e-mail
(async () => {
    const recipient = process.env.EMAIL_TO_NOTIFY;

    if (!recipient) {
        console.error("Erro: A variável de ambiente EMAIL_TO_NOTIFY não está configurada.");
        process.exit(1);
    }

    await sendEmail(recipient);
})();
